application/comit_node/src/swap_protocols/rfc003/messages.rs

```diff
@@ -0,0 +1,57 @@
+// TODO: Move things from comit_client::rfc003 to here
+use crate::swap_protocols::{
+    asset::Asset,
+    rfc003::{Ledger, SecretHash, SecretSource, Timestamp},
+    Metadata, RoleKind,
+};
+
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub struct Request<AL: Ledger, BL: Ledger, AA, BA> {
+    pub alpha_asset: AA,
+    pub beta_asset: BA,
+    pub alpha_ledger: AL,
+    pub beta_ledger: BL,
+    pub alpha_ledger_refund_identity: AL::Identity,
+    pub beta_ledger_redeem_identity: BL::Identity,
+    pub alpha_expiry: Timestamp,
+    pub beta_expiry: Timestamp,
+    pub secret_hash: SecretHash,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
+pub struct AcceptResponseBody<AL: Ledger, BL: Ledger> {
+    pub beta_ledger_refund_identity: BL::Identity,
+    pub alpha_ledger_redeem_identity: AL::Identity,
+}
+
+#[derive(Debug, Deserialize, PartialEq, Serialize)]
+pub struct RequestBody<AL: Ledger, BL: Ledger> {
+    pub alpha_ledger_refund_identity: AL::Identity,
+    pub beta_ledger_redeem_identity: BL::Identity,
+    pub alpha_expiry: Timestamp,
+    pub beta_expiry: Timestamp,
+    pub secret_hash: SecretHash,
+}
+
+impl<AL: Ledger, BL: Ledger, AA: Asset, BA: Asset> From<Request<AL, BL, AA, BA>> for Metadata {
+    fn from(request: Request<AL, BL, AA, BA>) -> Self {
+        Self {
+            alpha_ledger: request.alpha_ledger.into(),
+            beta_ledger: request.beta_ledger.into(),
+            alpha_asset: request.alpha_asset.into(),
+            beta_asset: request.beta_asset.into(),
+            role: RoleKind::Alice,
+        }
+    }
+}
+
+pub trait ToFullSwapRequest<AL: Ledger, BL: Ledger, AA: Asset, BA: Asset> {
```

Comment:
:thinking: Since the result of the method is named `Request`, maybe this should just be called `ToRequest`?